main/domoticz.cpp

```diff
@@ -854,12 +855,19 @@ int main(int argc, char**argv)
 				return 1;
 			}
 			std::string wwwport = cmdLine.GetSafeArgument("-sslwww", 0, "");
-			int iPort = (int)atoi(wwwport.c_str());
-			if ((iPort < 0) || (iPort > 32767))
+			int iPort = std::stoi(wwwport);
+			if ((iPort < 0) || (iPort > 65535))
```

Comment:
Could you change the maximum port range to 49151, these are 'user' ports and should stay this way.
Just forward another port like 44444 instead of 55555

Reply:
Yes, I can pick any port.
So now this PR is pretty minimal.

Reply:
It's unfraternally not minimal enough. There are more changes than just the two port ranges.
Using std::stoi will result in a (unhandled) crash when garbage was supplied (a non valid number, like a text)
Better keep it basic atoi

Reply:
std::atoi is C++
atoi is C
And indeed the exception throwing of `std::stoi` is different from `std::atoi`.

Reply:
This is a question on point-of-view. :-)
You need to compare with the current situation.
I have looked at the `atoi` use cases in this file and there is currently no safety net in place, if I understand the code correctly.
The problem with `atoi` https://en.cppreference.com/w/cpp/string/byte/atoi is:
> Return value
> Integer value corresponding to the contents of str on success.
> If no conversion can be performed, ​0​ is returned.
> If the value of the result cannot be represented, i.e. the converted value falls out of range of the corresponding return type, behavior is undefined. 

Undefined behavior means that the computer can blow up or a cat can die, for example.

More concretely, currently with `atoi` is:
```
./domoticz -loglevel  all -startupdelay toto
2022-01-12 20:46:48.702  Status: Domoticz V2021.1 (build 13959) (c)2012-2022 GizMoCuz
2022-01-12 20:46:48.702  Status: Build Hash: 00f9ca475, Date: 2022-01-01 12:59:09
2022-01-12 20:46:48.702  Status: Startup Path: /home/keryell/perso/Informatique/RaspberryPi/domoticz/
2022-01-12 20:46:48.702  Status: Startup delay... waiting 0 seconds...
```
or
```
./domoticz -loglevel  all -startupdelay 99999999999999999999999999999999999999999999999999999999999999999999999
2022-01-12 20:52:38.716  Status: Domoticz V2021.1 (build 13959) (c)2012-2022 GizMoCuz
2022-01-12 20:52:38.716  Status: Build Hash: 00f9ca475, Date: 2022-01-01 12:59:09
2022-01-12 20:52:38.716  Status: Startup Path: /home/keryell/perso/Informatique/RaspberryPi/domoticz/
2022-01-12 20:52:38.716  Status: Startup delay... waiting -1 seconds...
```
compared with `stoi`:
```
./domoticz -www 8080 -loglevel  all -startupdelay toto
2022-01-12 20:39:16.456  Status: Domoticz V2021.1 (build 13960) (c)2012-2022 GizMoCuz
2022-01-12 20:39:16.457  Status: Build Hash: 96b4c6766-modified, Date: 2022-01-11 09:09:45
2022-01-12 20:39:16.457  Status: Startup Path: /home/keryell/perso/Informatique/RaspberryPi/domoticz/
terminate called after throwing an instance of 'std::invalid_argument'
  what():  stoi
Aborted
```
or
```
/domoticz -loglevel  all -startupdelay 99999999999999999999999999999999999999999999999999999999999999999999999
2022-01-12 22:07:43.045  Status: Domoticz V2021.1 (build 13960) (c)2012-2022 GizMoCuz
2022-01-12 22:07:43.045  Status: Build Hash: 78bccf18b, Date: 2022-01-11 09:18:48
2022-01-12 22:07:43.045  Status: Startup Path: /home/keryell/perso/Informatique/RaspberryPi/domoticz/
terminate called after throwing an instance of 'std::out_of_range'
  what():  stoi
Aborted
```
I agree that it is not an ultimate solution for great UX design :-) ,  but at the end the `stoi` is safer since it avoids running a program with such an error by stopping immediately.
Anyway this code using the `atoi` is mostly about option parsing and thus could be updated to use instead a library like https://www.boost.org/doc/libs/1_78_0/doc/html/program_options.html I use on some of my projects https://github.com/keryell/BitWizard_SPI_relay/blob/master/examples/bw_spi_relay_control.cpp#L170 and which provides type safety, argument checking, help display...
But this is outside the scope of this PR. :-)
So what is the plan? Just keep all the `atoi` as the *status quo*?

Reply:
I favor going C++ (std::atoi) as that is the direction... so behavior stays the same (as plain atoi).

Domoticz already uses the Boost library. Rewriting the handling of program options to use this library so in the end things become even more stable, might be a good idea. But something for a separate PR I would say.

Reply:
This is getting way out of hand.
We keep the default atoi, and do we really want a higher port number ? If yes, I suggest to only add this change in this PR